channeldb/invoices.go

```diff
@@ -1604,7 +1673,32 @@ func fetchInvoice(invoiceNum []byte, invoices kvdb.RBucket) (Invoice, error) {
 
 	invoiceReader := bytes.NewReader(invoiceBytes)
 
-	return deserializeInvoice(invoiceReader)
+	invoice, err := deserializeInvoice(invoiceReader)
+	if err != nil {
+		return Invoice{}, err
+	}
+
+	// If this is an AMP invoice, then we'll also attempt to read out the
+	// set of HTLCs that were paid to prior set IDs. However, we'll only do
+	// this is the invoice didn't already have HTLCs stored in-line.
+	invoiceIsAMP := invoice.Terms.Features.HasFeature(
```

Comment:
Will there be a time when the payment hash will be omitted from AMP invoices?

Reply:
Good question...AFAIK rn it's actually _required_, and we just set _something_ right now, but it could very well be all zeroes. As is, we still use the payment hash as part of the indexing structure as well. 

For a smooth uptake of this feature i"d think we just want to keep including it, and maybe it can just eventually be some dummy value and we can detect an invoice of this nature using that instead.

Reply:
All zeroes would be useful to make things easier to understand for builders. Is it possible to keep using the hash as an index, but zero it out right before returning the invoice from `AddInvoice`? So that it becomes an implementation detail.

Reply:
FWIW, AMP invoices already have a bool on the RPC level (https://github.com/lightningnetwork/lnd/blob/ccc03931ad0c9e47006f845c7b2f60cd270be354/lnrpc/invoicesrpc/utils.go#L177), so developers can pick them out from normal invoices.

Reply:
Oh, I actually meant putting all zeroes in the bolt11 payment request - to signal that it is meaningless.